src/store/poll/utilities.test.js

```diff
@@ -0,0 +1,44 @@
+import defaultUtilities, * as utilities from './utilities'
+import { database } from '../../../firebase'
+
+afterAll(() => database.goOffline())
+
+describe('Utilities', () => {
+  describe('Generate hash', () => {
+    test('it only creates integers 1-4', () => {
+      const hashed = utilities.generateHash()
+      hashed
+        .toString()
+        .split('')
+        .map(str => parseInt(str, 10))
+        .forEach((num) => {
+          expect(num).toBeGreaterThanOrEqual(1)
+          expect(num).toBeLessThanOrEqual(4)
+        })
+    })
+    test('it generates a number', () =>
+      expect(typeof utilities.generateHash()).toBe('number'),
+    )
+    test('it generates a number of length four', () =>
+      expect(utilities.generateHash().toString().length).toBe(4),
+    )
+  })
+
+  describe('objectNotEmpty', () => {
+    test('it returns false if an object isn\'t empty', () => {
+      expect(utilities.objectNotEmpty({ a: true })).toBe(true)
+    })
+    test('it returns false if object is empty', () => {
+      expect(utilities.objectNotEmpty({})).toBe(false)
+    })
+  })
+
+  describe('getHash', () => {
```

Comment:
To truly test `getHash`, it seems like the key feature differentiating it from `generateHash` is that it checks against the db. Which means, we will need to check against a db - perhaps it would be sensible to set up a test db and populate `usedNums` (possibly needs a renaming to`usedCodes`/`usedKeys` when we change to strings).

A different (possibly easier and more reliable) approach is to mock the response of the db (set with all permutations of 4^4 combos) so we can make a reliable expectation about what `getHash` should return